src/OpenXRApi.cpp

```diff
@@ -418,34 +432,47 @@ OpenXRApi::OpenXRApi() {
 		return;
 	}
 
-	const bool SRGB_SWAPCHAIN = true;
+	// int64_t swapchainFormatToUse = swapchainFormats[0];
+	int64_t swapchainFormatToUse = 0;
+
+	// With the GLES2 driver we're rendering directly into this buffer with a pipeline that assumes an RGBA8 buffer.
+	// With the GLES3 driver rendering happens into an RGBA16F buffer with all rendering happening in linear color space.
+	// This buffer is then copied into the texture we supply here during the post process stage where tone mapping, glow, DOF, screenspace reflection and conversion to sRGB is applied. 
+	// As such we should chose an RGBA8 buffer here (note that an SRGB variant would allow automatic Linear to SRGB conversion but not sure if that is actually used)
 
-	int64_t swapchainFormatToUse = swapchainFormats[0];
+	// We grab the first applicable one we find, OpenXR sorts these from best to worst choice..
 
 	printf("Swapchain Formats\n");
-	for (int i = 0; i < swapchainFormatCount; i++) {
-		printf("%lX\n", swapchainFormats[i]);
+	for (int i = 0; i < swapchainFormatCount && swapchainFormatToUse == 0; i++) {
+		// printf("Found %llX\n", swapchainFormats[i]);
 #ifdef WIN32
-		if (SRGB_SWAPCHAIN && swapchainFormats[i] == GL_SRGB8_ALPHA8) {
+		if (swapchainFormats[i] == GL_SRGB8_ALPHA8) {
```

Comment:
I I don't know very well what godot does, but I do know when I choose a swapchain with `GL_RGBA8`, then the scene looks too bright here. But for now this works since GL_SRGB8_ALPHA8 is listed first here.

Reply:
The difference between an RGB and SRGB texture is that OpenGL will do automatic SRGB to Linear conversion on the SRGB texture format. As Godots final result is SRGB (as this is what we want to show on screen) and OpenXR likely expects linear color space as it applies its own color correction for display inside of the headset, having OpenGL internally convert back to linear color space solves the brightness issue.

I don't know however how much overhead is introduced here, it may well be better to use RGB and tell Godot not to do the linear -> SRGB color conversion.

Note also that if GLES2 is used there is no color conversion but all source images will be in SRGB color space anyway

Reply:
There is also some slight risk of loss of color precision seeing we're using 8bit per color textures here. Though since all the conversions internally are done on floating point values it should be ok.